### PR TITLE
Feature: SignAll - General case tx signing.

### PR DESCRIPTION
**Pull request by Tigh-Gherr**
Allows for signing the inputs of a tx in a more general way.

Example usage:
```go
func (s *signerService) Signer(ctx context.Context, lockingScript *bscript.Script) (Signer, error) {
	derivationPath := s.store.DerivationPath(ctx, lockingScript)

	derivKey, err := s.MasterPrivateKey.DeriveChildFromPath(derivationPath)
	if err != nil {
	        return nil, err
	}
	privKey, err := derivKey.ECPrivKey()
	if err != nil {
	        return nil, err
	}
	
	return &LocalSigner{PrivateKey: privKey}, nil
}

. . .

if err := tx.SignAll(ctx, &signerService{MasterPrivateKey: privKey}); err != nil {
	return nil, err
}
```

### CODE REVIEW

**Comment by jadwahab** [comment]
Not sure I follow, is this to derive a child key and sign in one step?

**Comment by Tigh-Gherr** [comment]
@jadwahab yeah, it's just to have a SignAuto function for derived keys, so that the implementor doesn't have to manage the building of them.

**Review by jadwahab — CHANGES_REQUESTED** [review]
Change SignAuto to Sign and rename SignAuto to SignAll

File: signer.go [hunk 1]
```diff
@@ -16,17 +16,3 @@ type Signer interface {
...
-// AutoSigner interface to allow custom implementations of different signing mechanisms.
-// Implement the Sign function as shown in InternalSigner, for example. Sign generates
-// and returns an ECDSA signature for the provided hash digest using the private key
-// as well as the public key corresponding to the private key used. The produced
-// signature is deterministic (same message and same key yield the same signature) and
-// canonical in accordance with RFC6979 and BIP0062.
-//
-// To automatically sign, the PublicKey() method must also be implemented in order to
-// use the public key to check which Inputs can be signed for before signing.
-type AutoSigner interface {
```
Comment (jadwahab) [inline signer.go:29]: peace the fuck out 😂

**Review by jadwahab — APPROVED** [review]

**Pull request merged**